### ISSUE TITLE
[Enhancement] RSS - Add video length to feed metadata

### ISSUE THREAD

**Issue by rbankosegger** [omarroth/invidious#3475]
**Is your enhancement request related to a problem? Please describe.**

I filter my RSS feeds with `grep` and similar tools. For example, I only keep posts where some phrase occurs in the `<title>` tag.
For video feeds, I would like to filter out videos that are too short (<30s) or too long (>2hrs).

**Describe the solution you'd like**

Within the `<entry>` tag of the feed, add a new tag, for example called `<yt:videoLengthInSeconds>` that has as its content the length of the current video.

**Additional context**

Other metadata might be worth considering too...
Only slightly related: I like the way `yt-dlp` handles metadata filtering with its `--match-filters` command line option. Maybe it is worth looking at for further metadata ideas and how they implement it. Citation from the manual:

> E.g.   `--match-filter  !is_live  --match-filter "like_count>?100  &  description~='(?i)&  dogs"` matches only videos that are not live OR those that have a like count more than 100 (or the like field is not available) and also has a description that contains the phrase  "cats  &  dogs"  (case‐less).

**Comment by altotin** [comment]
I would also find it useful to be able to filter locally on the video length.

Piped includes duration in its RSS: [example](https://pipedapi.kavin.rocks/feed/unauthenticated/rss?channels=UCLA_DiR1FfKNvjuUpBHmylQ)

Does Invidious have this metadata at the point RSS is generated?

**Comment by SamantazFox** [comment]
> Does Invidious have this metadata at the point RSS is generated?

No, we don't have that info available, as that would require making extra queries to youtube. At the moment, all the stuff in the ~RSS~ Atom feeds are what Youtube gives back to us.

**Comment by anakojm** [comment]
i think this feature should be added as this would allow to filter out shorts

**Comment by TheFrenchGhosty** [comment]
Not a feature on youtube... so this is kinda out of scope for Invidious
